app/services/quiz_service.py

```
@@ -0,0 +1,191 @@
+        await db.commit()
+
+        logger.info(f"Quiz created with id {new_quiz.id}")
+        return new_quiz
+
+    except SQLAlchemyError as e:
+        await db.rollback()
+        logger.error(f"Database error: {e}", exc_info=True)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail={"code": "DATABASE_ERROR", "message": "A database error occurred."},
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)